controllers/pattern_controller.go

```diff
@@ -153,6 +159,23 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if r.AnalyticsClient.SendPatternStartEventInfo(qualifiedInstance) {
 		return r.actionPerformed(qualifiedInstance, "Updated status with start event sent", nil)
 	}
+	var gitAuthSecret map[string][]byte
+	if qualifiedInstance.Spec.GitConfig.TokenSecret != "" {
+		if gitAuthSecret, err = r.authGitFromSecret(qualifiedInstance.Spec.GitConfig.TokenSecretNamespace, qualifiedInstance.Spec.GitConfig.TokenSecret); err != nil {
+			return r.actionPerformed(qualifiedInstance, "obtaining git auth info from secret", err)
+		}
+	}
+
+	gitDir := filepath.Join(qualifiedInstance.Status.LocalCheckoutPath, ".git")
+	if _, err = os.Stat(gitDir); os.IsNotExist(err) {
+		err = cloneRepo(r.gitOperations, qualifiedInstance.Spec.GitConfig.TargetRepo, qualifiedInstance.Status.LocalCheckoutPath, gitAuthSecret)
+		if err != nil {
+			return r.actionPerformed(qualifiedInstance, "cloning pattern repo", err)
+		}
+	}
+	if err = checkoutRevision(r.gitOperations, qualifiedInstance.Spec.GitConfig.TargetRepo, qualifiedInstance.Status.LocalCheckoutPath, qualifiedInstance.Spec.GitConfig.TargetRevision, gitAuthSecret); err != nil {
+		return r.actionPerformed(qualifiedInstance, "checkout target revision", err)
+	}
```

Comment:
Worth wrapping in a helper function to maintain readability of the Reconcile loop?